awx/main/tasks.py

```diff
@@ -339,15 +339,23 @@ def send_notifications(notification_list, job_id=None):
 def gather_analytics():
     if not settings.INSIGHTS_TRACKING_STATE:
         return
-    try:
-        tgz = analytics.gather()
-        if not tgz:
-            return
-        logger.debug('gathered analytics: {}'.format(tgz))
-        analytics.ship(tgz)
-    finally:
-        if os.path.exists(tgz):
-            os.remove(tgz)
+    last_time = settings.AUTOMATION_ANALYTICS_LAST_GATHER
+    gather_time = now()
+    if not last_time or ((gather_time - last_time) > timedelta(seconds=settings.AUTOMATION_ANALYTICS_GATHER_INTERVAL) * 0.9):
```

Comment:
Nitpicky, but I find this easier to parse:

`(gather_time - last_time).seconds > settings.AUTOMATION_ANALYTICS_GATHER_INTERVAL`

Reply:
Done, also removed the 0.9 fudge factor, since I'm not sure that will actually be needed in practice. If we end up sending every 4 hours + 5 minutes, it's ok.